src/utils/helpers/DrawHelpers.ts

```diff
@@ -0,0 +1,225 @@
+    return labelNodesGroup
+        .selectAll<HTMLElement, DependencyNode>('g#labels')
+        .data(nodes)
+        .enter()
+        .append<SVGGElement>('g')
+        .append('text')
+        .attr('fill', TextColors.DEFAULT)
+        .attr('cursor', 'default')
+        .text(d => d.name);
+}
+
+export function createMarkers(svgContainer: NodeSelection<SVGSVGElement>): void {
+    svgContainer
+        .append('svg:defs')
+        .append('svg:marker')
+        .attr('id', 'provider')
+        .attr('viewBox', '-5 -5 40 10')
+        .attr('refX', 15)
+        .attr('refY', 0)
+        .attr('markerWidth', 40)
+        .attr('markerHeight', 40)
+        .attr('orient', 'auto')
+        .append('svg:path')
+        .attr('d', 'M0,-5L20,0L0,5,q10 -5,0 -10')
+        .attr('fill', '#E5E5E6');
+}
+
+export function createSimulation(nodes: DependencyNode[], links: DependencyLink[], width: number, height: number) {
+    return forceSimulation(nodes)
+        .force(
+            'dependency',
+            forceLink<DependencyNode, DependencyLink>(links)
+                .distance(180)
+                .id((node: DependencyNode) => node.name)
+        )
+        .force('center', forceCenter(width / 2, height / 2))
+        .force('y', forceY(0.5))
+        .force('collide', forceCollide(140))
+        .force('nodeCollide', forceCollide(140));
+}
+
+export function createSVGContainer(
+    container: HTMLDivElement,
+    width: number,
+    height: number
+): Selection<SVGSVGElement, DependencyNode, Element, HTMLElement> {
+    return select<Element, DependencyNode>(`#${container.id}`)
+        .append('svg')
+        .attr('id', 'container')
+        .attr('preserveAspectRatio', 'xMidYMid meet')
+        .attr('viewBox', `0 0 ${width} ${height}`)
+        .attr('width', width)
+        .attr('height', height);
+}
+
+export function createZoom(svgContainer: NodeSelection<SVGSVGElement>): Selection<SVGGElement, DependencyNode, Element, HTMLElement> {
+    const zoomLayer = svgContainer.append('g').attr('id', 'zoom');
+
+    const zoomed = () => zoomLayer.attr('transform', event.transform);
+
+    svgContainer
+        .call(
+            zoom<SVGSVGElement, DependencyNode>()
+                .scaleExtent([1 / 2, 12])
+                .on('zoom', zoomed)
+        )
+        .on('dblclick.zoom', null);
+
+    return zoomLayer;
+}
+
+export function createLinkPath(this: Element, link: DependencyLink): void {
+    if (!link.source.x || !link.source.y || !link.target.x || !link.target.y) {
+        return;
+    }
+
+    const xDiff = link.source.x - link.target.x;
+    const yDiff = link.source.y - link.target.y;
+
+    const isSourceOnTheLeft = xDiff < 0;
+    const isSourceBelowTarget = yDiff > 0;
+
+    const angleInRadians = Math.abs(Math.atan(yDiff / xDiff));
+    const cosinus = Math.cos(angleInRadians);
+    const sinus = Math.sin(angleInRadians);
+
+    const offsetXLeft = -50 * cosinus;
+    const offsetY = 50 * sinus;
```

Comment:
Is 50 an arbitrary value?

Reply:
Yes, it is set based on subjective design feeling.